nf_core/modules/lint/main_nf.py

```diff
@@ -160,12 +167,23 @@ def check_process_section(self, lines):
         self.warned.append(("process_standard_label", "Process label unspecified", self.main_nf))
 
     for l in lines:
+        l = l.strip()
+        l = l.replace('"', "")
+        l = l.replace("'", "")
         if re.search("bioconda::", l):
             bioconda_packages = [b for b in l.split() if "bioconda::" in b]
-        if re.search("org/singularity", l):
-            singularity_tag = l.split("/")[-1].replace('"', "").replace("'", "").split("--")[-1].strip()
-        if re.search("biocontainers", l):
-            docker_tag = l.split("/")[-1].replace('"', "").replace("'", "").split("--")[-1].strip()
+        if l.startswith("https://containers") or l.startswith("https://depot"):
+            lspl = l.lstrip("https://").split(":")
+            if len(lspl) == 2:
+                ## e.g. 'https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img' :
+                singularity_tag = "_".join(lspl[0].split("/")[-1].strip().rstrip(".img").split("_")[1:])
```

Comment:
This stuff has regex written all over it (in my mind). I'll try to refactor when I get a moment, but equally if it works as it is then I'm happy for it to be merged and not to wait.

Reply:
I agree. Whenever you have time. It was tricky enough writing the Python code but a regex would be better.